docs/src/main/asciidoc/native-and-ssl.adoc

```diff
@@ -225,16 +225,24 @@ And let's build the native executable again:
 
 [WARNING]
 ====
-This behavior is new to GraalVM 19.3+.
+This behavior is new to GraalVM 21.3+.
 ====
 
-When creating a native binary, GraalVM embraces the principle of "immutable security" for the root certificates.
-This essentially means that the root certificates are fixed at image build time, based on the certificate configuration used at that build time
-(which for Quarkus means when you perform a build having `quarkus.package.type=native` set).
-This avoids shipping a `cacerts` file or requiring a system property be set in order to set up root
-certificates that are provided by the OS where the binary runs.
+GraalVM supports both build time and runtime certificate configuration.
+The build time approach favors the principle of "immutable security" where the appropriate certificates are added at build time, and can never be changed afterward.
+This guarantees that the list of valid certificates cannot be tampered with when the application gets deployed in production.
 
-As a consequence, system properties such as `javax.net.ssl.trustStore` do not have an effect at
+However, this comes with a few drawbacks:
+
+ * When a certificate expires, the application needs to be rebuilt, and redeployed into production with the new certificate, which is an inconvenience.
+ * Even worse, if a certificate gets revoked because of a security breach, all applications that embed this certificate need to be rebuilt and redeployed in a timely manner.
+ * Providing all certificates at build time complicates the CI, specifically in dynamic environments such as Kubernetes where valid certificates are provided by the platform in the `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt` PEM file.
+ * Lastly, this requires to add into the application all certificates for all environments (e.g. DEV, TEST, PROD), which means that a certificate that is required for DEV but should not be used elsewhere, will make its way anyway in production.
```

Comment:
Is this true? If you rebuild the application for each environment, it's not a problem right?

Reply:
if you do rebuild the app in each env, I agree it is not a problem. 
but we don't, and most work environments in which I worked don't either: software is built once and deployed in each env unchanged. otherwise what you deploy in production has never run anywhere. configuration changes, but the binary does not. at least that is what I have been used to. but I am sure this is a topic where people can have split positions.
what I can do is something like: _unless the application gets a dedicated built for each environment_ or _in case the decision is to run the exact same build in all environments_.

Reply:
see also discussion in https://github.com/oracle/graal/issues/1999

specifically https://github.com/oracle/graal/issues/1999#issuecomment-566522155, https://github.com/oracle/graal/issues/1999#issuecomment-589297911 and https://github.com/oracle/graal/issues/1999#issuecomment-566796334